tools/include/pass.hpp

```diff
@@ -24,17 +26,42 @@ struct pass
     /// A unique name used to identify the pass
     std::string name() const;
     /// Run the pass on the module
+    void apply(module_pass_manager& mpm) const;
     void apply(module& m) const;
```

Comment:
Same here ? 

Also curious why do we have the same logic both in `src/.../pass.hpp` and `tools/.../pass.hpp` ?

Reply:
> Also curious why do we have the same logic both in src/.../pass.hpp and tools/.../pass.hpp ?

The header under tools/include are ran through te.py to generate the type erasure code. This generated file is then placed under src/include. See [here](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/wiki/Developer's-Guide#type-erasure).